client/src/add_button/AddButton.js

```diff
@@ -0,0 +1,102 @@
+    const [weight, setWeight] = useState(0);
+    const [outByDate, setoutByDate] = useState(new Date());
+
+    const [showWeightWarning, setWeightWarning] = useState(false);
+    // Logic for adding recommendations will need to be added 
+    const [rec, setRec] = useState([]);
+    
+    return (
+        <div className="add_item_form_container">
+            <Form className="add_item_form">
+                <Form.Group className="input_field_container">
+                    <Form.Group>
+                        <Form.Label>Item Name</Form.Label>
+                        <Form.Control type="text" placeholder="Eg. Apples" onChange={event => setItemName(event.target.value.trim().toLowerCase())}/>
+                    </Form.Group>
+
+                    <b>OR</b>
+
+                    <Form.Group>
+                        <Form.Label>Item Number</Form.Label>
+                        <Form.Control type="text" placeholder="Eg. 1234" onChange={event => {setItemId(event.target.value.trim().toLowerCase())}}/>
+                    </Form.Group>
+
+                    <Form.Group>
+                        <Form.Label>Weight(lbs)</Form.Label>
+                        <Form.Control type="text" placeholder="Eg. 50" 
+                            onChange={event => {
+                                let val = Number(event.target.value.trim());
+
+                                // Checks for valid integer
+                                if (val !== Infinity && String(val) == val) {
+                                    setWeight(val);
+                                    setWeightWarning(false);
+                                }
+                                else {
+                                    setWeightWarning(true);
+                                }
+                            }
+                        }/>
+                        {showWeightWarning && <Form.Text className="warning">Invalid weight value</Form.Text>}
+                    </Form.Group>
+                    
+                    <Form.Group>
+                        <Form.Label>Out by date</Form.Label>
+                        <DatePicker selected={outByDate} onChange={date => setoutByDate(date)}/> 
+                    </Form.Group>
+
+                    <Form.Group>
+                        <NumericInput min = {0} size={1} value={lbPerHouse} onChange={val => setLbPerHouse(val)}/>
+                        lbs per Household
+                        <p id="household_calculator">= { (weight >= 0 && lbPerHouse > 0) ? weight / lbPerHouse : 0 } households </p>
+                    </Form.Group>
+
+                    <Form.Group>
+                        <div >
+                            Recommendations
+                            {(rec.length === 0)  ?
+                                <p className ="recommendation_pos">No Issues</p> :
+                                rec.map(txt=>{return <p className="recommendations_neg">{txt}</p>})
```

Comment:
I'm not too familiar with JSX, so I'm not sure if this is valid syntax. In general though, when an arrow function only returns a value, the curly braces and return can be replaced with the expression for the return value.

```suggestion
                        <div>
                            Recommendations
                            {(rec.length === 0)
                                ? <p className ="recommendation_pos">No Issues</p>
                                : rec.map(txt=>{return <p className="recommendations_neg">{txt}</p>})
```

Reply:
oh yeah, forgot about that syntax. Definitely cleaner though!